modin/utils.py

```diff
@@ -695,7 +697,7 @@ def get_current_execution() -> str:
     str
         Returns <StorageFormat>On<Engine>-like string.
     """
-    return f"{'Experimental' if IsExperimental.get() else ''}{StorageFormat.get()}On{Engine.get()}"
+    return f"{StorageFormat.get()}On{Engine.get()}"
```

Comment:
The factory name is now independent of the mode, since one factory should also provide entry points to experimental functionality so that they do not have to be switched when using functionality from different modes.

* For functions that do not exist in pandas, the entry points are separate namespaces (for methods: `DataFrame/Series.modin`, for free functions: `import modin.experimental.pandas`).
* For functions that exist in pandas and only have different implementations, switching implementations is done using environment variable `MODIN_EXPERIMENTAL` as before.

Reply:
Experimental IO classes have been removed as unnecessary.

Reply:
> We could probably combine some of them into one, and, I think since we have the executions such as ExpetimentalPyarrowOnRay and ExperimentalHdkOnNative, we should provide them when MODIN_EXPERIMENTAL is enabled, i.e. as is now.

@YarShev please note that for HDK the experimental mode is turned on automatically at the time of initialization.

Reply:
> The factory name is now independent of the mode, since one factory should also provide entry points to experimental functionality so that they do not have to be switched when using functionality from different modes.

What if we have experimental executions that are not initiatialized automatically as HDK or an external one?

> For functions that exist in pandas and only have different implementations, switching implementations is done using environment variable MODIN_EXPERIMENTAL as before.

> I think we should use MODIN_EXPERIMENTAL only for experimental executions and it should be decoupled from API. What if we just change function names that are present in pandas but only have different implementations?

> Experimental IO classes have been removed as unnecessary.

Could you elaborate this a bit why those are unnecesary?

Reply:
> What if we have experimental executions (exp pyarrow on ray, external one) that are not initiatialized automatically like HDK?

If we want to make the factory work only if the experimental mode is enabled, then we can put a mode check in the method that is used to initialize the factory - `def prepare(cls)`.

Reply:
> Could you elaborate this a bit why those are unnecesary?

All methods from experimental IO classes have been transferred to regular IO classes. And we moved them because these methods should be available in both modes, since now there are special namespaces through which the methods are **always** available. This is enough to understand the intent of the user who wants to use the experimental functionality.

Reply:
> Why do you think we shouldn't hold experimental factories that initialize respective executions?

The factory only routes calls, this task is the same for both modes, so I think we don’t need another one factory that differs only in name, but does the same job.

> Me: I think we should use MODIN_EXPERIMENTAL only for experimental executions and it should be decoupled from API. What if we just change function names that are present in pandas but only have different implementations?

Let's look at `groupby` example, we have an experimental (although it is activated by another environment variable, but still) and a main implementation. Do I understand correctly that your proposal suggests making a new `groupby_exp` function? In this case, there will be two functions with the same signatures (from the user’s point of view), and I believe there will be duplication of code in our implementation (albeit a little likely). In addition, now it’s enough to simply run user's code in two modes and see which is better from a performance point of view. If we make your change, anyone will need to make changes to the sources every time, which is quite inconvenient.